core/state/snapshot/generate.go

```diff
@@ -66,36 +66,40 @@ func (gs *generatorStats) Log(msg string, marker []byte) {
 			"at", common.BytesToHash(marker[common.HashLength:]),
 		}...)
 	}
-	// Calculate the estimated indexing time based on current stats
-	done := binary.BigEndian.Uint64(marker[:8]) - gs.origin
-	left := math.MaxUint64 - binary.BigEndian.Uint64(marker[:8])
-
-	speed := done / uint64(time.Since(gs.start)/time.Millisecond)
-	eta := time.Duration(left/speed) * time.Millisecond
-
-	log.Info(msg, append(ctx, []interface{}{
+	// Add the usual measurements
+	ctx = append(ctx, []interface{}{
 		"accounts", gs.accounts,
 		"slots", gs.slots,
 		"storage", gs.storage,
 		"elapsed", common.PrettyDuration(time.Since(gs.start)),
-		"eta", common.PrettyDuration(eta),
-	}...)...)
+	}...)
+	// Calculate the estimated indexing time based on current stats
+	if done := binary.BigEndian.Uint64(marker[:8]) - gs.origin; done > 0 {
+		left := math.MaxUint64 - binary.BigEndian.Uint64(marker[:8])
+
+		speed := done/uint64(time.Since(gs.start)/time.Millisecond+1) + 1 // +1s to avoid division by zero
+		ctx = append(ctx, []interface{}{
+			"eta", common.PrettyDuration(time.Duration(left/speed) * time.Millisecond),
```

Comment:
Could you add `percentage` too? I think that's a pretty user-friendly thing to have. Basically `100 * binary.BigEndian.Uint32(marker[:4]) / uint32(-1))`